### DISCUSSION TITLE
Next steps for your name, picture, and other info

### DISCUSSION THREAD

**Discussion by jaanus** [tact/public#94] [Ideas]
[Here’s a tweet that got me thinking.](https://twitter.com/mayli/status/1411576988175720456)

The somewhat anonymous nature of Tact means that you may end up with contacts with a vague name and picture. In the context of you connecting with them, you know who and why they are, but may forget.

So one thing that would well fit into the Tact concept, would be to have “private notes” for a contact. You can yourself just for our own reference enter their full name, how you know them, and anything else that might be relevant.

May-Li in that thread talks about more structured data like map etc. That might also be relevant. I think a simple private notes field would be a good start and could contain pointers to other data in link form, until we make this more structured, if we ever decide to go that route.

**Comment by jaanus** [comment]
After spending a while more thinking of these ideas, and also seeing through the launch of public beta how Tact works with a larger number of contacts, we have some new ideas and a plan for how we move forward with this.

## Coming next - private rename

We intentionally don’t restrict what name and picture you use to identify yourself in Tact. The social aspect of verifying identity also has not been a problem so far, as you start with connections with people whom you know anyway, who may be physically next to you as you make the connection.

This means, though, that you will end up with chats and people identified only with the first name or nickname. If there are several people with the same popular name, this may quickly get confusing.

So we’ll do **private rename**. You can rename a person to something that’s more clear to you. The renamed name is only visible to you (you can of course see also their original name that they picked for themselves). This will help you make the chat list more clear.

We have already committed to building private rename, and it will appear in Tact soon.

## Later - private notes and public bio

So after this, let’s go back to the original idea of private notes. Sometimes just rename is not enough, you’d want to add more private notes about someone about why/how you know them, as I described in my original message.

But then, isn’t it weird that you can have longer private notes for someone, but they can’t state anything about themselves? Yep - a bio. As is common on other social systems. **Public bio** and **private notes** nicely balance and complement each other, and we’ll likely build both together.

We haven‘t yet committed to building these, and are kicking around design ideas, might post some in this thread.

## Much later crazytalk - multiple profiles/avatars/identities

You only have one user account in Tact since you only can have one iCloud login on a device, and on Tact, you can currently have only one name and photo per user account, which is used in all 1:1 and group chats. This does not necessarily have to be so. You might want to vary and modulate how you are shown to others in a chat with family, work people, and some interest group.

We could build this in Tact so that you could have different identities in different groups, but we haven’t really thought much further in this direction yet.

**Comment by infotexture** [comment]
A private “alias” would be welcome, and the option to add notes would be a very useful next step.

Tweetbot does private notes well.

Here’s what the flow looks like on iOS, using the Tact Twitter account as an example:

![Tweetbot-Private-Notes](https://user-images.githubusercontent.com/129995/147552900-46fd90c6-cc2d-472d-989f-dabc179d2223.png)

**Reply by jaanus** [reply]
> Tweetbot does private notes well.

This is nuts. I have been using Tweetbot for 10+ years. I did not know they have this feature, until you said it here.

Yep. It works nicely there. Also has a UI bug seen on your last view, where the X button to delete the private note gets squished.
